modules/apps/static/portal-k8s-agent/portal-k8s-agent-impl/src/main/java/com/liferay/portal/k8s/agent/internal/osgi/commands/ClientExtensionsOSGiCommands.java

```diff
@@ -0,0 +1,275 @@
+			String[] parts = filterString.split("=", 2);
+
+			if (parts.length != 2) {
+				System.out.println("Invalid filter: " + filterString);
+
+				return null;
+			}
+
+			String key = parts[0];
+			String value = parts[1];
+
+			if (key.equals("deploymentType")) {
+				if (value.equals("agent")) {
+					deploymentFilterString = "(.k8s.config.key=*)";
+				}
+
+				if (value.equals("bundle")) {
+					deploymentFilterString = "(.cx.config.key=*)";
+				}
+
+				continue;
+			}
+
+			if (key.equals("webId")) {
+				key = "dxp.lxc.liferay.com.virtualInstanceId";
+
+				if (Objects.equals(PropsValues.COMPANY_DEFAULT_WEB_ID, value)) {
+					value = "default";
+				}
+			}
+
+			sb.append("(");
+			sb.append(key);
+			sb.append("=");
+			sb.append(value);
+			sb.append(")");
+		}
+
+		return _configurationAdmin.listConfigurations(
+			StringBundler.concat("(&", deploymentFilterString, sb, ")"));
+	}
+
+	private String _getConfigurationTableRow(
+		Configuration configuration, String formatString) {
+
+		Dictionary<String, Object> properties = configuration.getProperties();
+
+		return String.format(
+			formatString, configuration.getPid(), properties.get("name"),
+			properties.get("type"),
+			properties.get("dxp.lxc.liferay.com.virtualInstanceId"));
+	}
+
+	private String _printConfiguration(Configuration configuration) {
+		return StringBundler.concat(
+			"\nPID: ", configuration.getPid(), "\nFactoryPID: ",
+			configuration.getFactoryPid(), "\nBundle location: ",
+			configuration.getBundleLocation(), "\n",
+			_formatProperties(configuration.getProperties()));
+	}
+
+	private void _printConfigurations(Configuration[] configurations) {
+		int idWidth = 150;
+		int nameWidth = 40;
+		int typeWidth = 20;
+		int webIdWidth = 15;
+
+		String formatString = StringBundler.concat(
+			"| %-", idWidth, "s | %-", nameWidth, "s | %-", typeWidth, "s | %-",
+			webIdWidth, "s |%n");
+
+		System.out.printf(formatString, "pid", "name", "type", "webId");
+
+		int totalWidth =
+			idWidth + nameWidth + typeWidth + webIdWidth + (4 * 3) + 2;
+
+		System.out.println("-".repeat(totalWidth));
+
+		for (Configuration configuration : configurations) {
+			System.out.println(
+				_getConfigurationTableRow(configuration, formatString));
+		}
+	}
+
+	private Configuration _reloadConfiguration(Configuration configuration)
+		throws IOException {
+
+		Dictionary<String, Object> originalProperties =
+			configuration.getProperties();
+
+		String originalPid = configuration.getPid();
+
+		configuration.delete();
+
+		try (SafeCloseable safeCloseable =
+				InMemoryOnlyConfigurationThreadLocal.
+					setInMemoryOnlyWithSafeCloseable(true)) {
+
+			Configuration reloadedConfiguration =
+				ConfigurationUtil.getConfiguration(
+					_configurationAdmin, originalPid);
+
+			reloadedConfiguration.update(originalProperties);
+
+			return reloadedConfiguration;
+		}
+		catch (Exception exception) {
+			throw new RuntimeException(exception);
+		}
```

Comment:
I think we can just allow this method to `throw` directly. What advantage do we gain by catching and re-throwing as a `RuntimeException`?